internal/monitors/kubernetes/cluster/clusterstate.go

```diff
@@ -19,20 +19,22 @@ import (
 // State makes use of the K8s client's "reflector" helper to watch the API
 // server for changes and keep the datapoint cache up to date,
 type State struct {
-	clientset  *k8s.Clientset
-	reflectors map[string]*cache.Reflector
-	namespace  string
-	cancel     func()
+	clientset                  *k8s.Clientset
+	reflectors                 map[string]*cache.Reflector
+	namespace                  string
+	nodeConditionTypesToReport []string
+	cancel                     func()
 
 	metricCache *metrics.DatapointCache
 }
 
-func newState(clientset *k8s.Clientset, metricCache *metrics.DatapointCache, namespace string) *State {
+func newState(clientset *k8s.Clientset, metricCache *metrics.DatapointCache, namespace string, nodeConditionTypesToReport []string) *State {
```

Comment:
Pass the `nodeConditionTypesToReport` list into the constructor of the DatapointCache at https://github.com/signalfx/signalfx-agent/blob/master/internal/monitors/kubernetes/cluster/monitor.go#L126 (like `useNodeName`) so that you don't have to pass it down through this function.